pkg/pod/pod.go

```diff
@@ -168,6 +171,15 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 		}
 	}
 
+	// Add env var if hermetic execution was requested
+	if taskRun.Spec.ExecutionMode.Hermetic {
+		for i, s := range stepContainers {
```

Comment:
I think for now this alpha feature can only cover step containers, since the sidecar containers aren't controlled by the entrypointer (https://github.com/tektoncd/pipeline/issues/1569) -- without the entrypointer I'm not sure how to enforce a hermetic environment on the sidecar containers. 

If anyone has any suggestions pls let me know!

Reply:
Oh I see!  Good point.

Reply:
Ugh, I was hoping the sidecar KEP would be started by now but it looks like it missed the 1.22 train as well. A couple suggestions in the meantime:

* do nothing, document that sidecars are not hermetic
* disallow sidecars when hermetic mode is enabled

I can see pros and cons of each, let's start with 1. for alpha and get some feedback?